crates/broker/src/order_picker.rs

```diff
@@ -259,7 +301,7 @@ where
 
         if skip_preflight {
             // If we skip preflight we lockin the order asap
-            return Ok(Some(OrderLockTiming { target_timestamp_secs: 0, expiry_secs: expiration }));
+            return Ok(Lock { target_timestamp_secs: 0, expiry_secs: expiration });
```

Comment:
We wouldn't be able to lock a lock expired order, so we need to check if its lock expired or not

Reply:
added extra conditional here